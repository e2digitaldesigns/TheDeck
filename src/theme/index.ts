export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
