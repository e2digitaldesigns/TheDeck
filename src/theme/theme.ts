export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
