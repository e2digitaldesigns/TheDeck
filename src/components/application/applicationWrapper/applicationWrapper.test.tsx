import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

import ApplicationWrapper from "./applicationWrapper";
import { ThemeProvider } from "styled-components";
import { theme } from "../../../theme/theme";

const setup = () => {
  return render(
    <ThemeProvider theme={theme}>
      <ApplicationWrapper />
    </ThemeProvider>
  );
};

describe("<ApplicationWrapper />", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("Should render correctly", () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
