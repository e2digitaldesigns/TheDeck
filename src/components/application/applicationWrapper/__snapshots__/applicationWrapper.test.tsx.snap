// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApplicationWrapper /> Should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        data-testid="template-wrapper-section"
      >
        <div
          class="Toastify"
        />
        <header
          class="sc-bTfYFJ dcxWKJ"
          data-testid="template_header__component"
        >
          <ul
            class="sc-kHOZwM bePSHX"
            data-testid="template_header__menu"
          >
            <li>
              <a
                href="#/home"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="#/macroDeck"
              >
                MacroDeck
              </a>
            </li>
            <li>
              <a
                href="#/settings"
              >
                Settings
              </a>
            </li>
          </ul>
          <ul
            class="sc-kHOZwM sc-hOGkXu bePSHX iEmzpZ"
            data-testid="template_header__options"
          >
            <li
              data-testid="template_header__minimize-button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 13H5v-2h14v2z"
                />
              </svg>
            </li>
            <li
              data-testid="template_header__fs-button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
                />
              </svg>
            </li>
            <li
              data-testid="template_header__close-button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </li>
          </ul>
        </header>
        <div
          class="sc-dtMgUX ehLuqf"
        >
          <img
            alt="macro deck logo"
            src="logo.png"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      data-testid="template-wrapper-section"
    >
      <div
        class="Toastify"
      />
      <header
        class="sc-bTfYFJ dcxWKJ"
        data-testid="template_header__component"
      >
        <ul
          class="sc-kHOZwM bePSHX"
          data-testid="template_header__menu"
        >
          <li>
            <a
              href="#/home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="#/macroDeck"
            >
              MacroDeck
            </a>
          </li>
          <li>
            <a
              href="#/settings"
            >
              Settings
            </a>
          </li>
        </ul>
        <ul
          class="sc-kHOZwM sc-hOGkXu bePSHX iEmzpZ"
          data-testid="template_header__options"
        >
          <li
            data-testid="template_header__minimize-button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 13H5v-2h14v2z"
              />
            </svg>
          </li>
          <li
            data-testid="template_header__fs-button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
              />
            </svg>
          </li>
          <li
            data-testid="template_header__close-button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </li>
        </ul>
      </header>
      <div
        class="sc-dtMgUX ehLuqf"
      >
        <img
          alt="macro deck logo"
          src="logo.png"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
